src/BitPaySDK/Client.php

```diff
@@ -289,62 +291,141 @@ public function getInvoices(
     }
 
     /**
-     * Create a BitPay refund.
+     * Create a refund for a BitPay invoice.
      *
-     * @param $invoice          Invoice A BitPay invoice object for which a refund request should be made.  Must have
-     *                          been obtained using the merchant facade.
-     * @param $refundEmail      string The email of the buyer to which the refund email will be sent
-     * @param $amount           float The amount of money to refund. If zero then a request for 100% of the invoice
-     *                          value is created.
-     * @param $currency         string The three digit currency code specifying the exchange rate to use when
-     *                          calculating the refund bitcoin amount. If this value is "BTC" then no exchange rate
-     *                          calculation is performed.
-     * @return bool True if the refund was successfully canceled, false otherwise.
-     * @throws BitPayException BitPayException class
+     * @param $invoiceId          The BitPay invoice Id having the associated refund to be created.
+     * @param $amount             Amount to be refunded in the currency indicated.
+     * @param $currency           Reference currency used for the refund, usually the same as the currency used to create the invoice.
+     * @param $preview            Whether to create the refund request as a preview (which will not be acted on until status is updated)
+     * @param $immediate          Whether funds should be removed from merchant ledger immediately on submission or at time of processing
+     * @param $buyerPaysRefundFee Whether the buyer should pay the refund fee (default is merchant)
+     * @return Refund             An updated Refund Object
+     * @throws RefundCreationException RefundCreationException class
+     * @throws BitPayException    BitPayException class
      */
     public function createRefund(
-        Invoice $invoice,
-        string $refundEmail,
+        string $invoiceId,
         float $amount,
-        string $currency
-    ): bool {
-        try {
-            $refund = new Refund($refundEmail, $amount, $currency, $invoice->getToken());
-            $refund->setGuid(Util::guid());
+        string $currency,
+        bool $preview,
```

Comment:
initialise optional fields, here all booleans to false
e.g: bool $preview = false,